src/RecordManager/Base/Record/Lido.php

```diff
@@ -237,10 +237,22 @@ public function getTitle(
         $allTitles = [];
         foreach ($this->getTitleSetNodes() as $set) {
             foreach ($set->appellationValue as $appellationValue) {
-                if ($lang == null || $appellationValue['lang'] == $lang) {
-                    $titles[] = (string)$appellationValue;
+                if (!($title = trim((string)$appellationValue))) {
+                    continue;
+                }
+                $allTitles[] = $title;
+                if (!$lang) {
+                    continue;
+                }
+                $priority = (string)$appellationValue['pref'];
+                $titleLang = (string)$appellationValue['lang'];
+                if (('preferred' === $priority && $titleLang === $lang)
```

Comment:
I think you could simplify the conditions quite a bit by giving these defaults (also attributes are always strings, so no casting needed, right?):
```
priority = $appellationValue['pref'] ?? 'preferred';
$titleLang = $appellationValue['lang'] ?? $lang;
```

Reply:
Casting is a habit when working with SimpleXmlElements... Will adjust.

Reply:
...And I was wrong, the cast is needed.

Reply:
Updated the example to reflect that.